lib/compiler/passes/generate-bytecode.js

```diff
@@ -502,6 +517,10 @@ function generateBytecode(ast) {
           action: null
         }),
         buildCondition(
+          // Check expression match, not the node match
+          // If expression always match, no need to replace FAILED to NULL,
+          // because FAILED will never appeared
+          -(node.expression.match | 0),
```

Comment:
why is this random code repeated all over the place?

why isn't it behind a method with a clear name that explains what it does?

do you believe this comment tells a reader what you're doing?

Reply:
Explained above.

Reply:
No, it wasn't.  You gave a Zorba the Mystic "when you look deep enough into the soul of `match` you will understand."